### PR TITLE
Add explanation for Session Create, Delete, List commands

### PR DESCRIPTION
**Pull request by froststein**

### CODE REVIEW

File: docs/DeveloperGuide.md [hunk 1]
```diff
@@ -219,8 +219,41 @@ when the user invokes the `session /create` command.
 <br>
 <br>
 The general workflow of the `session /create` command is as follows:
+1. The user input provided is passed to `SplitLah`.
+2. `SplitLah` then parses the input by using methods in the `Parser` class to obtain a `SessionCreateCommand` object.
+3. A `SessionCreateCommand#run` method is then invoked to run the `session /create` command.
```
Comment (matheril) [inline docs/DeveloperGuide.md:224]: The "a" might be unnecessary here, since other lines don't have it.

File: docs/DeveloperGuide.md [hunk 2]
```diff
@@ -219,8 +219,41 @@ when the user invokes the `session /create` command.
...
+when the user invokes the `session /delete` command.
+<br>
+<br>
+![Delete Session Sequence Diagram Screenshot](https://raw.githubusercontent.com/AY2122s2-cs2113t-t10-1/tp/master/docs/images/developerguide/SessionDeleteCommand.drawio.png)
+<br>
+<br>
+The general workflow of the `session /delete` command is as follows:
+1. The user input provided is passed to `SplitLah`.
+2. `SplitLah` then parses the input by using methods in the `Parser` class to obtain a `SessionDeleteCommand` object.
+3. `SessionDeleteCommand#run` method is then invoked to run the `session /delete` command.
```
Comment (matheril) [inline docs/DeveloperGuide.md:245]: Add "the" at the start

File: docs/DeveloperGuide.md [hunk 3]
```diff
@@ -219,8 +219,41 @@ when the user invokes the `session /create` command.
...
+4. The list of sessions are stored in a `Profile` object, hence `Manager#getProfile` is called
+   before the list of sessions can be retrieved.
+5. Once the `Profile` object is returned, `Profile#getSession` is called to retrieve the session with the specified 
+session unique identifier from the list of sessions.
+   1. If the specified session unique identifier cannot be found, a `String` object 
+   representing session could not be found is returned.
+   2. Else the `Session` object with the specified session unique identifier is returned.
+6. With the `Session` object, it is removed from the list of sessions stored in `Profile` object.
```
Comment (matheril) [inline docs/DeveloperGuide.md:253]: This line should probably be rephrased, it doesn't sound right.

**Review by matheril — CHANGES_REQUESTED** [review]
Some phrasing changes recommended

**Review by matheril — APPROVED** [review]
Looks good to merge

File: docs/DeveloperGuide.md [hunk 4]
```diff
@@ -219,8 +219,41 @@ when the user invokes the `session /create` command.
 <br>
 <br>
 The general workflow of the `session /create` command is as follows:
+1. The user input provided is passed to `SplitLah`.
+2. `SplitLah` then parses the input by using methods in the `Parser` class to obtain a `SessionCreateCommand` object.
+3. `SessionCreateCommand#run` method is then invoked to run the `session /create` command.
+4. Once the command starts to run, `SessionCreateCommand` class checks if there is an existing session with the same session name.
+5. If an existing session with the specified session name is found, a message indicating that another session exists with the same name is printed using `TextUi#printlnMessage`.
```
Comment (ivanaitzliddat) [inline docs/DeveloperGuide.md:226]: Perhaps can rephrase it as "a message indicating that another session with the same name exists is printed..."

File: docs/DeveloperGuide.md [hunk 5]
```diff
@@ -233,18 +266,19 @@ when the user invokes the `session /list` command.
...
-     calling `TextUi#printlnMessage()` to print out a brief overview of each session.
-     Then, the last group is printed with a divider below it, using the method `TextUi#printlnMessageWithDivider()`.
+4. The list of sessions are stored in a `Profile` object, hence `Manager#getProfile` is called
+   before the list of sessions can be retrieved.
+5. The `SessionListCommand` object runs the `Profile#getSessionListSummaryString` method to retrieve a `String` object
+   representing the summaries of the sessions stored.
+    1. If there are no sessions stored in the `Profile` object, a `String` object representing an empty list of sessions is
+       returned.
+    2. Else, the `Profile` objects instantiates a new `TableFormatter` object and loops through the list of sessions,
+       calling `TableFormatter#addRow` for each group to create a table with the summary of each session. A `String` object
```
Comment (ivanaitzliddat) [inline docs/DeveloperGuide.md:279]: Perhaps can change each group to each session

**Review by ivanaitzliddat — CHANGES_REQUESTED** [review]
Looks good overall, possible to edit the changes requested

File: docs/DeveloperGuide.md [hunk 6]
```diff
@@ -219,8 +219,41 @@ when the user invokes the `session /create` command.
...
+2. `SplitLah` then parses the input by using methods in the `Parser` class to obtain a `SessionDeleteCommand` object.
+3. The `SessionDeleteCommand#run` method is then invoked to run the `session /delete` command.
+4. The list of sessions are stored in a `Profile` object, hence `Manager#getProfile` is called
+   before the list of sessions can be retrieved.
+5. Once the `Profile` object is returned, `Profile#getSession` is called to retrieve the session with the specified 
+session unique identifier from the list of sessions.
+   1. If the specified session unique identifier cannot be found, a `String` object 
+   representing an error message is returned.
+   2. Else the `Session` object with the specified session unique identifier is returned.
+6. The returned `Session` object is then removed from the list of sessions stored in `Profile` object.
```
Comment (warrencxw) [inline docs/DeveloperGuide.md:253]: Perhaps can mention how is it removed, e.g. using `Profile#removeSession` method or something along those lines.

**Review by warrencxw — CHANGES_REQUESTED** [review]
Several minor errors found in the documentation. I have added a few suggestions, please reference them and edit accordingly.

**Pull request merged**